File: C#/Palindrome/Palindrome.cs
using System;
using System.Linq;

namespace PalindromeProg
{
    /// <summary>
    /// 
    /// </summary>
    static class Palindrome
    {
        /// <summary>
        /// 
        /// </summary>
        /// <param name="strInput"></param>
        public static void TestPalindrome( string strInput )
        {
            var rslt = Check( strInput );
            Console.WriteLine( string.Format( "{0} {1}", strInput, rslt ) );
        }

        /// <summary>
        /// 
        /// </summary>
        /// <param name="strInput"></param>
        /// <returns></returns>
        public static bool Check( string strInput )
        {
            // Remove all spaces and set to lowercase.
            strInput = strInput.Replace( " ", "" ).ToLower();
            // Remove all punctuation
            strInput = new string( strInput.Where( c => !char.IsPunctuation( c ) ).ToArray() );

            //Console.WriteLine( strInput );
            int strlen = strInput.Length;
            // Check length.
            if ( strlen < 3 )
            {
                return false;
            }

            int i = 0,
                j = strlen - 1;
            while ( i < j )
            {
                //Console.WriteLine( "{0}, {1}", i, j );
                if ( strInput[i] != strInput[j] )
                {
                    return false;
                }

                ++i;
                --j;
            }

            return true;
        }
    }
}

File: Javascript/Palindrome/palindrome.js
const palindrome_check = (str_input) => {
  if (typeof str_input != "string") {
    return false;
  }

  // Convert to lowercase, remove punctuation,
  // and make into list of characters.
  str_input = str_input
    .toLowerCase()
    .replace(/[^a-z0-9]/g, "")
    .split("");

  // Get and check length.
  str_length = str_input.length;
  if (str_length < 3) {
    return false;
  }

  i = 0;
  j = str_length - 1;
  while (i < j) {
    //console.log(i, j);
    if (str_input[i] != str_input[j]) {
      return false;
    }
    ++i;
    --j;
  }

  return true;
};

module.exports = { palindrome_check };
